pkg/alertmanager/operator.go

```diff
@@ -654,6 +658,21 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	level.Info(logger).Log("msg", "sync alertmanager")
 
+	if c.canReadStorageClass {
+		if !reflect.DeepEqual(am.Spec.Storage.VolumeClaimTemplate, monitoringv1.EmbeddedPersistentVolumeClaim{}) {
+			storageClassName := am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName
+			if storageClassName != nil && *storageClassName != "" {
+				_, err := c.kclient.StorageV1().StorageClasses().Get(ctx, *storageClassName, metav1.GetOptions{})
+				if err != nil {
```

Comment:
That's a lot of nesting for an if-block. See if you can refactor it to make the code more readable. If not, then atleast put some comments explaining the 2nd and 3rd if conditions (line 662 and 664).

Reply:
Sure will refactor it

Reply:
Updated